metric2/instrument/instrument.go

```diff
@@ -0,0 +1,4 @@
+package instrument
```

Comment:
Could this live in the `metric` package instead?

Reply:
There's a cycle if I do this. I'll try and move things around, thanks for pointing this out.

Reply:
I have restructured the draft API so that the top-level `metric` package contains the `Instrument` and `Measurement` types, with sub-package `meter`, `meter/asyncint64`, `meter/asyncfloat64`, `meter/syncint64`, and `meter/syncfloat64`. I'm using a type named `Builder` for the instrument constructors, to make it clear that you only need to know sync/async and int64/float64 when building instruments, not when recording measurements.